lib/minify.js

```diff
@@ -101,6 +102,7 @@ function minify(files, options) {
         if (options.ie) set_shorthand("ie", options, [ "compress", "mangle", "output", "rename" ]);
         if (options.keep_fargs) set_shorthand("keep_fargs", options, [ "compress", "mangle", "rename" ]);
         if (options.keep_fnames) set_shorthand("keep_fnames", options, [ "compress", "mangle", "rename" ]);
+        if (options.module) set_shorthand("module", options, [ "compress", "parse" ]);
```

Comment:
you want `"mangle"` here too, and the appropriate handling in `lib/scope.js`.

Reply:
Oh? I had a quick glance and didn't notice anything under `mangle` that relies on `"use strict";` &minus; what did I miss? 🤔

Reply:
And in case you meant `toplevel` &minus; it is toggled by `module` under
https://github.com/mishoo/UglifyJS/blob/8e664ca647cdec10e6e68cfc4a76146ea4804b8f/lib/minify.js#L91

Reply:
Yes, toplevel mangle.  That's probably sufficient.

Reply:
The included test does cover this somewhat:
```js
$ cat test/input/module/input.js
console.log(function() {
    function sum(...params) {
        return this || arguments[0] + arguments[1];
    }
    return sum(sum(1, 3), 5);
}());
console.log(function() {
    "use strict";
    function sum(...params) {
        return this || arguments[0] + arguments[1];
    }
    return sum(sum(2, 4), 6);
}());
```
```js
$ uglifyjs test/input/module/input.js --module -bcm
function n() {
    return this || arguments[0] + arguments[1];
}

function o() {
    return this || arguments[0] + arguments[1];
}

console.log(n(n(1, 3), 5)), console.log(o(o(2, 4), 6));
```
```js
$ uglifyjs test/input/module/input.js --module -bcm toplevel=false
function a() {
    return this || arguments[0] + arguments[1];
}

function c() {
    return this || arguments[0] + arguments[1];
}

console.log(a(a(1, 3), 5)), console.log(c(c(2, 4), 6));
```
(Note: `sum()`s have been `rename`d as they weren't top-level functions, but `mangle` did not subsequently act on the `compress`ed top-level ones)